cmd/dataset_tokenizer/dataset_tokenizer.go

```diff
@@ -476,9 +476,9 @@ func (tt TextsTokenizer) TokenizeTexts(
 // Consumes a ContextsIterator function and serializes the contexts to an
 // aligned binary file.
 func WriteContexts(outPath string, nextContext ContextsIterator,
-	encoder *gpt_bpe.GPTEncoder) (int, error) {
+	encoder *gpt_bpe.GPTEncoder, sampling int, sortSpec string) (int, error) {
```

Comment:
There's only one `sortSpec` that we use, `shuffle`, so this should be a `shuffle bool`.